operator/controllers/btpoperator_controller.go

```diff
@@ -232,20 +234,20 @@ func (r *BtpOperatorReconciler) HandleProcessingState(ctx context.Context, cr *v
 	ready, err := manifest.InstallChart(&logger, installInfo, []types.ObjectTransform{r.labelTransform})
 	if err != nil {
 		logger.Error(err, fmt.Sprintf("error while installing resource %s", client.ObjectKeyFromObject(cr)))
-		return r.UpdateBtpOperatorState(ctx, cr, types.StateError)
+		return r.UpdateBtpOperatorStatus(ctx, cr, types.StateError, ChartInstallFailed, fmt.Sprintf("error while installing resource %s", client.ObjectKeyFromObject(cr)))
 	}
 	if ready {
-		return r.UpdateBtpOperatorState(ctx, cr, types.StateReady)
+		return r.UpdateBtpOperatorStatus(ctx, cr, types.StateReady, ReconcileSucceeded, "Reconcile succeeded")
 	}
 
 	return nil
 }
 
 func (r *BtpOperatorReconciler) HandleDeletingState(ctx context.Context, cr *v1alpha1.BtpOperator) error {
 	logger := log.FromContext(ctx)
-	if err := r.handleDeprovisioning(ctx); err != nil {
+	if err := r.handleDeprovisioning(ctx, cr); err != nil {
 		logger.Error(err, "deprovisioning failed")
-		return r.UpdateBtpOperatorState(ctx, cr, types.StateError)
+		return err
```

Comment:
returning error will keep BtpOperator in `Deleting` state and trigger Reconcile again

Reply:
This value is either `nil` or the result of `UpdateBtpOperatorStatus` call. It is never the error from any other function.